### PR TITLE
Increase TextUtil test coverage

### PR DESCRIPTION
**Pull request by shouxian92**
Addresses #388 

1. There is a simple logic fix on [#L337](https://github.com/eeverman/andhow/compare/master...shouxian92:increase-textutil-coverage?expand=1#diff-8a18ca74b03a03e942869877a3675e39L337) and [#L359](https://github.com/eeverman/andhow/compare/master...shouxian92:increase-textutil-coverage?expand=1#diff-8a18ca74b03a03e942869877a3675e39L359) of `TextUtil.java`, since my assumption is that the method should return some index only if both input parameters `toBeSearched` and `toBeFound` are present
2. Adds a line to test unicode characters on `escapeXml()`
3. Added a test which checks for strings that goes into "multiline" when it is beyond the string length's limit for `println()` function

### CODE REVIEW

**Comment by eeverman** [comment]
Thank you @shouxian92 - I will need to take closer look at your changes before merging - More then I can do on my phone :-)
I should be able to look this evening - Thanks again.

File: andhow-core/src/test/java/org/yarnandtail/andhow/util/TextUtilTest.java [hunk 1]
```diff
@@ -233,4 +283,24 @@ public void testNullToEmpty() {
...
+
+	@Test
+	public void testFindFirstInstanceOfWithEmptyToBeFound() {
+		assertEquals(-1, TextUtil.findFirstInstanceOf("abcd", 10));
+	}
+
+	@Test
+	public void testFindLastInstanceOfWithEmptyToBeFound() {
+		assertEquals(-1, TextUtil.findLastInstanceOf("abcd", 10));
+	}
```
Comment (eeverman) [inline andhow-core/src/test/java/org/yarnandtail/andhow/util/TextUtilTest.java]: These are good tests for the null and empty cases, but I think we need some tests for normal cases of find first and find last.

File: andhow-core/src/main/java/org/yarnandtail/andhow/util/TextUtil.java [hunk 2]
```diff
@@ -324,7 +324,9 @@ public static boolean toBoolean(String value) {
 	
 	/**
 	 * Find the first occurrence of one of an array of possible strings in another string,
-	 * starting at the specified position.
+	 * starting at the specified position, looking to the right. If there are more than one
+	 * string arguments to be searched from, it returns the index of the argument which is
+	 * of the lower index
```
Comment (eeverman) [inline andhow-core/src/main/java/org/yarnandtail/andhow/util/TextUtil.java]: Awesome!  Thank you for the improved docs!

File: andhow-core/src/test/java/org/yarnandtail/andhow/util/TextUtilTest.java [hunk 3]
```diff
@@ -233,4 +285,59 @@ public void testNullToEmpty() {
...
 	}
+
+	@Test
+	public void testFindFirstInstanceOf() {
+		assertEquals(31, TextUtil.findFirstInstanceOf(FIND_INSTANCE_STRING_BROWNFOX, 3, "the", "lazy"));
+		assertEquals(40, TextUtil.findFirstInstanceOf(FIND_INSTANCE_STRING_BROWNFOX, 6, "dog"));
+		assertEquals(20, TextUtil.findFirstInstanceOf(FIND_INSTANCE_STRING_BROWNFOX, -1, "jumps", "over"));
+		assertEquals(0, TextUtil.findFirstInstanceOf(FIND_INSTANCE_STRING_BROWNFOX, 0, "The", "fox"));
+		assertEquals(65, TextUtil.findFirstInstanceOf(FIND_INSTANCE_STRING_BROWNFOX, 30, "jumps", "over"));
+		assertEquals(49, TextUtil.findFirstInstanceOf(FIND_INSTANCE_STRING_BROWNFOX, 30, "brown", "quick"));
```
Comment (eeverman) [inline andhow-core/src/test/java/org/yarnandtail/andhow/util/TextUtilTest.java]: Nice 'found' additions.

**Comment by eeverman** [comment]
Great Work on this, @shouxian92 
Let me know if there is a project I can help you with.

**Comment by eeverman** [comment]
This was some nice work, @shouxian92 - Would you like to be added as a collaborator on the AndHow project?

**Comment by shouxian92** [comment]
Thanks for extending the invitation @eeverman but i'm currently not looking to commit to any projects at the moment. I'm glad that you thought so highly of me! 😄

**Pull request merged**